packages/@aws-cdk/aws-lambda-nodejs/lib/function.ts

```diff
@@ -73,6 +73,13 @@ export interface NodejsFunctionProps extends lambda.FunctionOptions {
    *   modules are bundled.
    */
   readonly bundling?: BundlingOptions;
+
+  /**
+   * The path to the directory containing project config files (`package.json` or `tsconfig.json`)
+   *
+   * @default - the directory containing the `depsLockFilePath`
+   */
+  readonly projectRoot?: string;
```

Comment:
This prop is not passed to `Bundling.bundle()`, it has currently no effect.

Also I think we need to validate that `props.depsLockFilePath` in located inside `props.projectRoot`

Reply:
> This prop is not passed to `Bundling.bundle()`, it has currently no effect.

Fixed

> Also I think we need to validate that `props.depsLockFilePath` in located inside `props.projectRoot`

Generally I think this will be true but is it a hard requirement?

Reply:
nice to have...